src/main/java/serguei/http/HttpResponse.java

```diff
@@ -19,9 +19,15 @@
     private final long contentLength;
     private final boolean chunked;
 
-    public HttpResponse(InputStream inputStream) throws IOException {
+    public HttpResponse(InputStream inputStream, HttpRequestHeaders requestHeaders) throws IOException {
         this.headers = new HttpResponseHeaders(inputStream);
-        contentLength = headers.getContentLength();
+
+        if (requestHeaders != null && requestHeaders.getAllowedContentLength() != -1) {
+            this.contentLength = requestHeaders.getAllowedContentLength();
+        } else {
+            this.contentLength = headers.getContentLength();
+        }
+
```

Comment:
What happens when we reach contentLength with the connection? Is it just left hanging there with the next request on the connection attempting to read whatever is left from the previous request instead of the headers?